src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs

```diff
@@ -288,20 +302,256 @@ private static ProjectItemGroupElement CreateItemGroup(Project project, string f
+                    Strings.Error_AssetsFileNotFound,
+                    project.FullPath));
+                return null;
+            }
+
+
+            var lockFileFormat = new LockFileFormat();
+            var assetsFile = lockFileFormat.Read(assetsPath);
+
+            //Packages from the assets file
+            var requestedVersions = GetRequestedPackages(project, userInputFrameworks, assetsFile);
+
+            //Packages resolved versions from the assets file including transitive
+            //packages if include-transitive flag is used
+            var packagesPerFramework = GetResolvedVersions(userInputFrameworks, assetsFile, requestedVersions, transitive);
+
+            return packagesPerFramework;
+        }
+
+        /// <summary>
+        /// Given a project, a list of frameworks and an assets file, this function returns all the
+        /// target frameworks with their packages
+        /// </summary>
+        /// <param name="project">PR project file</param>
+        /// <param name="userInputFrameworks">List of frameworks</param>
+        /// <param name="assetsFile">Assets file</param>
+        /// <returns></returns>
+        private Dictionary<string, IEnumerable<PRPackage>> GetRequestedPackages(Project project, IList<string> userInputFrameworks, LockFile assetsFile)
+        {
+            var result = new Dictionary<string, IEnumerable<PRPackage>>();
+
+            foreach (var tfm in assetsFile.PackageSpec.TargetFrameworks)
+            {
+                if (userInputFrameworks.Count == 0 || userInputFrameworks.Contains(tfm.ToString())) //If tfm is in the list given by the user or no frameworks were specified
+                {
+                    var projPackages = GetPackageReferencesPerFramework(project, "", tfm.ToString());
+
+                    result.Add(tfm.ToString(), tfm.Dependencies.Select(d => {
+
+                        
+                        string versionStr; //requested version exactly as it is in the proj file
+                        VersionRange version; //requested version value as a range
+                        //If the package is not auto-referenced, get the version from the project file. Otherwise fall back on the assets file
+                        if (!d.AutoReferenced)
+                        {
+                            versionStr = projPackages.Where(p => p.EvaluatedInclude.Equals(d.Name)).First().GetMetadataValue(VERSION_TAG);
+                            version = VersionRange.Parse(versionStr);
+                        }
+                        else
+                        {
+                            versionStr = d.LibraryRange.VersionRange.ToString();
+                            version = d.LibraryRange.VersionRange;
+                        }
+
+                        return new PRPackage {
+                            name = d.Name,
+                            requestedVersion = version,
+                            projectFileRequestedVersion = versionStr,
```

Comment:
Why do you really need `requestedVersion` and `projectFileRequestedVersion`? As I mentioned earlier, you just need RequestedVersionStr.